### PR TITLE
Adding species-level flavours and refactoring how reagent flavour is handled.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Makes `nutriment_desc` and `taste_description` both write to a field of reagent data, and then clear themselves.
- Encodes mixed taste data to JSON and only decodes when necessary for mixing or displaying taste information to a user.
- Allows taste strings to be set up as associative lists with species names for per-species taste information.
- Updates many strings to have Teshari-specific lines, as the original goal of this monstrosity was just letting Tesh not taste sweet things.

### CODE REVIEW

**Comment by Atermonera** [comment]
Wait that's the wrong  button

**Comment by Cerebulon** [comment]
The taste strings you have changed seem fine, but it seems like for Drinks you made "spicy" things taste different while for Food you only changed sweet things and left spicy things spicy. I don't mind either way, but it should probably be consistent tasting/not tasting Spice unless it was meant to just be differing interpretations of "strong alcohol" in which case hot noodles (drink) is an outlier.

**Comment by Cerebulon** [comment]
Proceed With The Spicelessness

**Review by Atermonera — DISMISSED** [review]

File: code/modules/projectiles/projectile/arc.dm [hunk 1]
```diff
@@ -191,7 +191,7 @@
 				var/obj/effect/vfx/water/splash = new(T)
 				splash.create_reagents(15)
 				splash.reagents.add_reagent("stomacid", 5)
-				splash.reagents.add_reagent("blood", 10,list("blood_colour" = "#ec4940"))
+				splash.reagents.add_reagent("blood", 10,TASTE_DATA(list("blood_colour" = "#ec4940")))
```
Comment (Spookerton) [inline code/modules/projectiles/projectile/arc.dm:194]: taste data?

File: code/modules/reagents/reagent_containers/blood_pack.dm [hunk 2]
```diff
@@ -36,7 +36,7 @@
 	if(blood_type != null)
 		label_text = "[blood_type]"
 		update_iv_label()
-		reagents.add_reagent(reag_id, 200, list("donor"=null,"viruses"=null,"blood_DNA"=null,"blood_type"=blood_type,"resistances"=null,"trace_chem"=null))
+		reagents.add_reagent(reag_id, 200, TASTE_DATA(list("donor"=null,"viruses"=null,"blood_DNA"=null,"blood_type"=blood_type,"resistances"=null,"trace_chem"=null)))
```
Comment (Spookerton) [inline code/modules/reagents/reagent_containers/blood_pack.dm:39]: taste data?

File: code/modules/xenoarcheaology/effects/sweating_reagents.dm [hunk 3]
```diff
@@ -29,7 +29,7 @@
 	spawn()
 		var/obj/effect/vfx/water/splash = new(T)
 		splash.create_reagents(15)
-		splash.reagents.add_reagent("blood", 10, list("blood_colour" = effect_color))
+		splash.reagents.add_reagent("blood", 10, TASTE_DATA(list("blood_colour" = effect_color)))
```
Comment (Spookerton) [inline code/modules/xenoarcheaology/effects/sweating_reagents.dm:32]: taste data?

**Review by Spookerton — CHANGES_REQUESTED** [review]

**Review by Spookerton — APPROVED** [review]

**Pull request merged**